src/witness/circom.rs

```diff
@@ -49,38 +54,43 @@ impl Circom for Wasm {
 }
 
 #[cfg(feature = "circom-2")]
-impl Circom2 for Wasm {
+impl Circom2 for WasmInstance {
     fn get_field_num_len32(&self) -> Result<u32> {
         self.get_u32("getFieldNumLen32")
     }
 
     fn get_raw_prime(&self) -> Result<()> {
         let func = self.func("getRawPrime");
-        func.call(&[])?;
+        let mut store = self.store.write().unwrap();
```

Comment:
Why not?
```suggestion
        let mut store = self.store.read().unwrap();
```
Same for all "getters"